packages/ui/src/theme/tokens/components/badge.ts

```diff
@@ -1,43 +1,38 @@
-import {
-  BackgroundColorValue,
-  ColorValue,
-  DesignToken,
-  FontSizeValue,
-  FontWeightValue,
-  LineHeightValue,
-  RadiusValue,
-  SpaceValue,
-  TextAlignValue,
-} from '../types/designToken';
+import { DesignTokenProperties } from '../types/designToken';
 
-interface BadgeVariationTokens {
-  backgroundColor: DesignToken<BackgroundColorValue>;
-  color: DesignToken<ColorValue>;
-}
+type VariationKey = 'backgroundColor' | 'color';
 
-interface BadgeSizeTokens {
-  fontSize: DesignToken<FontSizeValue>;
-  paddingHorizontal: DesignToken<SpaceValue>;
-  paddingVertical: DesignToken<SpaceValue>;
-}
+type BadgeVariationTokens<OutputType> = DesignTokenProperties<
+  VariationKey,
+  OutputType
+>;
 
-export interface BadgeTokens {
-  backgroundColor: DesignToken<BackgroundColorValue>;
-  borderRadius: DesignToken<RadiusValue>;
-  color: DesignToken<ColorValue>;
-  error: BadgeVariationTokens;
-  fontSize: DesignToken<FontSizeValue>;
-  fontWeight: DesignToken<FontWeightValue>;
-  info: BadgeVariationTokens;
-  large: BadgeSizeTokens;
-  lineHeight: DesignToken<LineHeightValue>;
-  paddingHorizontal: DesignToken<SpaceValue>;
-  paddingVertical: DesignToken<SpaceValue>;
-  small: BadgeSizeTokens;
-  success: BadgeVariationTokens;
-  textAlign: DesignToken<TextAlignValue>;
-  warning: BadgeVariationTokens;
-}
+type SizeKey = 'fontSize' | 'paddingHorizontal' | 'paddingVertical';
+
+type BadgeSizeTokens<OutputType> = DesignTokenProperties<SizeKey, OutputType>;
+
+type BadgeKey =
+  | 'backgroundColor'
+  | 'borderRadius'
+  | 'color'
+  | 'fontSize'
+  | 'fontWeight'
+  | 'lineHeight'
+  | 'paddingHorizontal'
+  | 'paddingVertical'
+  | 'textAlign';
+
+export type BadgeTokens<OutputType = unknown> = DesignTokenProperties<
+  BadgeKey,
+  OutputType
+> & {
+  error?: BadgeVariationTokens<OutputType>;
+  large?: BadgeSizeTokens<OutputType>;
+  small?: BadgeSizeTokens<OutputType>;
+  info?: BadgeVariationTokens<OutputType>;
+  success?: BadgeVariationTokens<OutputType>;
+  warning?: BadgeVariationTokens<OutputType>;
```

Comment:
Setting the properties in the base `BadgeTokens` will allow us to remove the usage of `PartialDeep` on `Tokens`, not sure if that's super important or anything but thought it would be nice to have